Makefile

```diff
@@ -0,0 +1,62 @@
+# This Makefile captures common developer build and test use cases.
+
+MAKEFLAGS += --no-print-directory
+
+# - incrementally compile and install the c++ code
+# - copy the shared objects on top of the python dev build
+.PHONY: update
```

Comment:
Linux error:

```
$ make update
make[1]: /tmp/pip-build-env-imfu1ozp/overlay/local/lib/python3.10/dist-packages/cmake/data/bin/cmake: No such file or directory
make[1]: *** [Makefile:180: cmake_check_build_system] Error 127
make: *** [Makefile:9: update] Error 2
```

MacOS error:

```
johnkerl@Kerl-MBP[prod][gspowley/sc-22584/build-flow-improvements][TileDB-SOMA]$ make update
[ 50%] Built target ep_spdlog
[ 56%] Performing configure step for 'libtiledbsoma'
-- Using default install prefix /Users/johnkerl/git/single-cell-data/TileDB-SOMA/dist. To control CMAKE_INSTALL_PREFIX, set OVERRIDE_INSTALL_PREFIX=OFF
-- Install prefix is /Users/johnkerl/git/single-cell-data/TileDB-SOMA/dist.
-- Skipping search for TileDB, building it as an external project. To use system TileDB, set FORCE_BUILD_TILEDB=OFF
-- Starting TileDB-SOMA regular build.
-- Performing Test HAVE_AVX2
-- Performing Test HAVE_AVX2 - Success
-- Starting TileDB-SOMA build.
-- Found TileDB: TILEDB_LIB-NOTFOUND
-- Building with commit hash 4ba0d1e
CMake Warning at src/CMakeLists.txt:57 (message):
  Building TileDB without deprecation warnings


CMake Error at src/pyapi/CMakeLists.txt:3 (find_package):
  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "/Users/johnkerl/git/single-cell-data/TileDB-SOMA/build/libtiledbsoma/CMakeFiles/CMakeOutput.log".
See also "/Users/johnkerl/git/single-cell-data/TileDB-SOMA/build/libtiledbsoma/CMakeFiles/CMakeError.log".
make[3]: *** [libtiledbsoma-prefix/src/libtiledbsoma-stamp/libtiledbsoma-configure] Error 1
make[2]: *** [CMakeFiles/libtiledbsoma.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [update] Error 2
```

Note this happened on MacOS even after `pip install -r apis/python/requirements_dev.txt`